### PR TITLE
V1.4 add tests

### PR DESCRIPTION
**Pull request by nbvannhi**
Tests for #247 
- [X] logic/parsers/ShowCommandParser
- [x] model/student/Assessment
- [x] model/student/AssessmentList
- [x] model/student/Score

### CODE REVIEW

File: src/test/java/seedu/address/logic/parser/ShowCommandParserTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        // parse by prefixes
+        assertParseSuccess(parser, NAME_DESC_AMY, new ShowCommand(student.getName(), null));
+        assertParseSuccess(parser, ID_DESC_AMY, new ShowCommand(student.getId(), null));
+        assertParseSuccess(parser, ASSESSMENT_DESC_AMY, new ShowCommand(assessment, null));
+        assertParseSuccess(parser, GROUP_DESC_TUTORIAL, new ShowCommand(group, null));
+    }
+
+    @Test
+    public void parse_presentPath() {
```
Comment (jonas-chow) [inline src/test/java/seedu/address/logic/parser/ShowCommandParserTest.java:61]: I believe you need to add a `+ PREFIX_FILE` at the end of the inputs for these tests? Also seems like the notion of equality in ShowCommand doesn't include savePath

File: src/test/java/seedu/address/model/student/AssessmentListTest.java [hunk 2]
```diff
@@ -50,88 +55,119 @@ public void add_nullAssessment_throwsNullPointerException() {
...
+        assertThrows(NullPointerException.class, () -> assessments.remove(null));
     }
 
     @Test
-    public void setAssessment_editedAssessmentHasDifferentIdentity_success() {
-        assessments.add(PATH_01);
-        assessments.setAssessment(PATH_01, MISSION_01);
-        AssessmentList expectedUniqueAssessmentList = new AssessmentList();
-        expectedUniqueAssessmentList.add(MISSION_01);
+    public void remove_personDoesNotExist_throwsAssessmentNotFoundException() {
```
Comment (jonas-chow) [inline src/test/java/seedu/address/model/student/AssessmentListTest.java]: Please rename the method to `assessmentDoesNotExist`

**Review by jonas-chow — CHANGES_REQUESTED** [review]
Apart from those things I pointed out in the comments, LGTM.

File: src/test/java/seedu/address/logic/parser/ShowCommandParserTest.java (hunk 1 above)
Comment (nbvannhi) [reply src/test/java/seedu/address/logic/parser/ShowCommandParserTest.java:61]: True ;-; no wonder why the tests still passed thanks!

**Review by jonas-chow — APPROVED** [review]
LGTM

**Pull request merged**